test/suite/test_checkpoint_snapshot06.py

```diff
@@ -0,0 +1,250 @@
+        ds_1.populate()
+
+        ds_2 = SimpleDataSet(self, self.uri_2, 0, \
+                key_format=self.key_format, value_format=self.value_format, \
+                config="log=(enabled=true)" + self.extraconfig)
+        ds_2.populate()
+
+        # Insert number of records into both tables.
+        self.large_updates(self.uri_1, self.valuea, ds_1, self.nrows, 0)
+        self.check(self.valuea, self.uri_1, self.nrows, 0, False)
+
+        self.large_updates(self.uri_2, self.valuea, ds_2, self.nrows, 0)
+        self.check(self.valuea, self.uri_2, self.nrows, 0, False)
+
+        # Remove one key from both the tables.
+        cursor1 = self.session.open_cursor(self.uri_1)
+        cursor2 = self.session.open_cursor(self.uri_2)
+
+        cursor1.set_key(ds_1.key(50))
+        cursor2.set_key(ds_2.key(50))
+        self.assertEqual(cursor1.remove(), 0)
+        self.assertEqual(cursor2.remove(), 0)
+
+        # Truncate the range from 1 - 100 where the key 50 doesn't exist.
+        session1 = self.conn.open_session()
+        session1.begin_transaction()
+        cursor11 = session1.open_cursor(self.uri_1)
+        cursor12 = session1.open_cursor(self.uri_2)
+
+        cursor11.set_key(ds_1.key(100))
+        cursor12.set_key(ds_2.key(100))
+        session1.truncate(None, None, cursor11, None)
+        session1.truncate(None, None, cursor12, None)
+
+        # Insert the key 50 that was already removed.
+        session2 = self.conn.open_session()
+        session2.begin_transaction()
+        cursor21 = session2.open_cursor(self.uri_1)
+        cursor22 = session2.open_cursor(self.uri_2)
+
+        cursor21.set_key(ds_1.key(50))
+        cursor21.set_value(self.valueb)
+        cursor22.set_key(ds_2.key(50))
+        cursor22.set_value(self.valueb)
+
+        self.assertEqual(cursor21.insert(), 0)
+        self.assertEqual(cursor22.insert(), 0)
+
+        # Create a checkpoint thread
+        done = threading.Event()
+        ckpt = checkpoint_thread(self.conn, done)
+        try:
+            ckpt.start()
+
+            # Wait for checkpoint to start and acquire its snapshot before committing.
+            ckpt_snapshot = 0
+            while not ckpt_snapshot:
+                time.sleep(1)
+                stat_cursor = self.session.open_cursor('statistics:', None, None)
+                ckpt_snapshot = stat_cursor[stat.conn.txn_checkpoint_snapshot_acquired][2]
+                stat_cursor.close()
+
+            # commit the operations in out of order. Insert followed by truncate.
+            session2.commit_transaction()
+            session1.commit_transaction()
+
+            # Evict all the modifications of the table1 before checkpoint gets into the table.
+            self.evict_cursor(self.uri_1, ds_1, self.nrows)
+
+        finally:
+            done.set()
+            ckpt.join()
+
+        # Perform additional checkpoint to ensure that the table2 has also have the latest data.
```

Comment:
Minor grammar fixes. I suggest:
Perform an additional checkpoint to ensure table2 also has the latest data.

Reply:
Fixed as suggested.